lib/ansible/modules/crypto/luks_device.py

```diff
@@ -0,0 +1,448 @@
+            return False
+
+        if self._module.params['device'] is not None:
+            name = self.get_container_name_by_device(
+                self._module.params['device'])
+            # sucessfully getting name based on device means that luks is open
+            luks_is_open = name is not None
+
+        if self._module.params['name'] is not None:
+            device = self.get_container_device_by_name(
+                self._module.params['name'])
+            # sucessfully getting device based on name means that luks is open
+            luks_is_open = device is not None
+
+        return luks_is_open
+
+    def luks_add_key(self):
+        if (self._module.params['device'] is None or
+                self._module.params['key'] is None or
+                self._module.params['new_key'] is None):
+            # conditions for adding a key not fulfilled
+            return False
+
+        if self._module.params['state'] == 'absent':
+            self._module.fail_json(msg="Contradiction in setup: Asking to "
+                                   "add a key to absent LUKS.")
+
+        return True
+
+    def luks_remove_key(self):
+        if (self._module.params['device'] is None or
+                self._module.params['remove_key'] is None):
+            # conditions for removing a key not fulfilled
+            return False
+
+        if self._module.params['state'] == 'absent':
+            self._module.fail_json(msg="Contradiction in setup: Asking to "
+                                   "remove a key from absent LUKS.")
+
+        return True
+
+    def luks_remove(self):
+        return (self._module.params['device'] is not None and
+                self._module.params['state'] == 'absent' and
+                self.is_luks(self._module.params['device']))
+
+
+def run_module():
+    # available arguments/parameters that a user can pass
+    module_args = dict(
+        state=dict(type='str', choices=['present', 'absent'], required=False, default='present'),
+        device=dict(type='str', required=False),
+        open=dict(type='bool', required=False, default=False),
+        name=dict(type='str', required=False),
+        key=dict(type='path', required=False),
+        new_key=dict(type='path', required=False),
+        remove_key=dict(type='path', required=False)
+    )
+
+    # seed the result dict in the object
+    result = dict(
+        changed=False,
+        name=None
+    )
+
+    module = AnsibleModule(argument_spec=module_args,
+                           supports_check_mode=False)
+
+    crypt = CryptHandler(module)
+    conditions = Conditions(module)
+
+    # The conditions are in order to allow more operations in one run.
+    # (e.g. create luks and add a key to it)
+
+    # luks create
+    if conditions.luks_create():
+        try:
+            crypt.run_luks_create(module.params['device'],
+                                  module.params['key'])
+        except ValueError as e:
```

Comment:
Why not have a large `try: except` block around all the operations which catches `ValueError`? That would improve readability alot.